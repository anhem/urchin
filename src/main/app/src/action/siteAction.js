import {Actions} from '../constants'

export const toggleNavigation = (navigationVisible) => (dispatch) => {
    return dispatch({
        type: Actions.Site.TOGGLE_NAVIGATION,
        data: !navigationVisible
    });
};