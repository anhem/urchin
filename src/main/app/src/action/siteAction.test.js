import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import fetchMock from 'fetch-mock'
import {toggleNavigation} from "./siteAction";

describe('userAction', () => {

    let store;

    beforeEach(() => {
        store = configureMockStore([thunk])({});
    });

    afterEach(() => {
        fetchMock.reset();
        fetchMock.restore();
    });

    it('toggleNavigation is successful', () => {

        store.dispatch(toggleNavigation(false));
        expect(store.getActions()).toMatchSnapshot();
    });

});