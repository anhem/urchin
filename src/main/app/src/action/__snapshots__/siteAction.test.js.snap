// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userAction toggleNavigation is successful 1`] = `
Array [
  Object {
    "data": true,
    "type": "TOGGLE_NAVIGATION",
  },
]
`;
